From 9b591272975871168a146a25c0a4d98b498129f0 Mon Sep 17 00:00:00 2001
From: hari gowtham <hgowtham@redhat.com>
Date: Mon, 2 Nov 2015 15:21:19 +0530
Subject: [PATCH 84/98] correction of message displayed after attach tier

the message after attach tier is saying rebalance.
It is changed according to tiering.

Change-Id: I1834511f86483fa60f404d7defe5be59c025e9d6
BUG: 1275925
Signed-off-by: hari gowtham <hgowtham@redhat.com>
Tested-by: Gluster Build System <jenkins@build.gluster.com>
Reviewed-by: Dan Lambright <dlambrig@redhat.com>
Tested-by: Dan Lambright <dlambrig@redhat.com>
Signed-off-by: hari <hgowtham@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/60884
---
 cli/src/cli-rpc-ops.c |   45 +++++++++++++++++++++++++++++++++++----------
 1 files changed, 35 insertions(+), 10 deletions(-)

diff --git a/cli/src/cli-rpc-ops.c b/cli/src/cli-rpc-ops.c
index 0de43d0..64ca5f7 100644
--- a/cli/src/cli-rpc-ops.c
+++ b/cli/src/cli-rpc-ops.c
@@ -1745,12 +1745,25 @@ gf_cli_defrag_volume_cbk (struct rpc_req *req, struct iovec *iov,
                                  * case since unlock failures can be highlighted
                                  * event though rebalance command was successful
                                  */
-                                snprintf (msg, sizeof (msg),
-                                          "Rebalance on %s has been started "
-                                          "successfully. Use rebalance status "
-                                          "command to check status of the "
-                                          "rebalance process.\nID: %s\n%s",
-                                          volname, task_id_str, rsp.op_errstr);
+
+                                 if (cmd == GF_DEFRAG_CMD_START_TIER) {
+                                         snprintf (msg, sizeof (msg),
+                                                  "Attach tier is successful "
+                                                  "on %s. use tier status to "
+                                                  "check the status.\nID: %s"
+                                                  "\n%s",
+                                                  volname, task_id_str,
+                                                  rsp.op_errstr);
+                                 } else {
+                                         snprintf (msg, sizeof (msg),
+                                                  "Rebalance on %s has been "
+                                                  "started successfully. Use "
+                                                  "rebalance status command to"
+                                                  " check status of the "
+                                                  "rebalance process.\nID: %s\n%s",
+                                                  volname, task_id_str,
+                                                  rsp.op_errstr);
+                                 }
                          } else {
                                 snprintf (msg, sizeof (msg),
                                           "Starting rebalance on volume %s has "
@@ -1823,11 +1836,23 @@ done:
                                     rsp.op_errstr);
         else {
                 if (rsp.op_ret)
-                        cli_err ("volume rebalance: %s: failed: %s", volname,
-                                 msg);
+                        if (cmd == GF_DEFRAG_CMD_START_TIER) {
+                                cli_err ("Tiering Migration Functionality: %s:"
+                                         " failed%s%s", volname,
+                                         strlen (msg) ? ": " : "", msg);
+                        } else
+                                cli_err ("volume rebalance: %s: failed%s%s",
+                                         volname, strlen (msg) ? ": " : "",
+                                         msg);
                 else
-                        cli_out ("volume rebalance: %s: success: %s", volname,
-                                 msg);
+                        if (cmd == GF_DEFRAG_CMD_START_TIER) {
+                                cli_out ("Tiering Migration Functionality: %s:"
+                                         " success%s%s", volname,
+                                         strlen (msg) ? ": " : "", msg);
+                        } else
+                                cli_out ("volume rebalance: %s: success%s%s",
+                                         volname, strlen (msg) ? ": " : "",
+                                         msg);
         }
         ret = rsp.op_ret;
 
-- 
1.7.1

